uber/site_sections/preregistration.py

```diff
@@ -524,3 +527,22 @@ def shirt_reorder(self, session, message='', **params):
             'message': message,
             'attendee': attendee
         }
+
+    def apply_promo_code(self, session, message='', **params):
+        if 'id' and 'code' in params:
+            if params['code'] is not '':
+                attendee, group = self._get_unsaved(params['id'])
+                if not attendee:
```

Comment:
This essentially means that you can never apply a promo code to a group, because you check only for the existence of a single attendee. You probably don't want this.